libsolidity/interface/CompilerStack.cpp

```diff
@@ -114,9 +115,10 @@ void CompilerStack::setSource(string const& _sourceCode)
 bool CompilerStack::parse()
 {
 	//reset
+	if(m_stackState < SourcesSet)
+		return false;
 	m_errors.clear();
 	ASTNode::resetID();
-	m_parseSuccessful = false;
```

Comment:
Hmm, apparently one could have called `parser()` repeatedly which only reset the AST nodes, but not the compilation states?